pandas/core/frame.py

```diff
@@ -5980,6 +5980,49 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
                     var_name=var_name, value_name=value_name,
                     col_level=col_level)
 
+    def explode(self, col_name, sep=None, dtype=None):
+        """
+        Create a new DataFrame where each element in each row
+        of a list-like column `col_name` is expanded to its own row
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        col_name : str
+            Name of the column to be exploded
+        sep : str, default None
+            Convenience to split a string `col_name` before exploding
+        dtype : str or dtype, default None
+            Optionally coerce the dtype of exploded column
+-
+        Examples
```

Comment:
Perhaps add a See Also linking to `Series.str.split`, `Series.str.extract`? Maybe others?

Are we interested in implementing the inverse operation (what dplyr calls `unite`: https://tidyr.tidyverse.org/reference/unite.html)?

Reply:
Good idea. I'll add Series.str.split and Series.str.extract here. Which other ones do you think would be relevant?

`unite`: so it would be like a groupby.agg(list/concat) type of operation? I'm not opposed to it but I think there's no urgency since we haven't had much user demand. My guess is because it maps to groupby so it's more natural to think about than the reverse.